cosmoz-grouped-list.js

```diff
@@ -231,12 +235,32 @@
 				});
 
 				if (group.items.length) {
-					return flatData.concat(group, group.items);
+					return acc.concat(group, group.items);
 				} else if (this.displayEmptyGroups) {
-					return flatData.concat(group);
+					return acc.concat(group);
 				}
-				return flatData;
+				return acc;
 			}, []);
+
+			this.selectedItems = this.selectedItems.filter(i => flatData.includes(i));
+
+			return flatData;
+		}
+
+		_assertDataIsNotMixed(data) {
```

Comment:
As the comment below, avoid negating things, so maybe rename it "_assertDataIsCoherent" ?
Otherwise statements like `if (!dataIsNotMixed) { .. }` will blow our minds :)